### PR TITLE
dada2: Bugs in macros.xml

### PR DESCRIPTION
**Pull request by pmpcosta**
FOR CONTRIBUTOR:
* [ ] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [ ] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [X] - This PR does something else (explain below)

Bug squashing.

read.table was correcting names to turn them into syntactically valid variable names. As a result samples names could be altered. in my use case, sample names start with a number, which I assume is quite common, so, for example, sample 31i was renamed X31i. This affects at least removeBimeraDenovo and sequence counts. The result is a scrambled sequence count table.

I also found a typo in the read.uniques function. I can't see what it might affect, but it looks rather dire.

### CODE REVIEW

**Comment by bernt-matthias** [comment]
We also need to bump tool versions.

File: tools/dada2/macros.xml [hunk 1]
```diff
@@ -38,17 +38,17 @@ echo $(R --version | grep version | grep -v GNU)", dada2 version" $(R --vanilla
          - uniques is a named integer vector (columns=ASVs, only one rows)-->
     <token name="@READ_FOO@"><![CDATA[
     read.uniques <- function ( fname ) {
-         p <- read.table(fname, header=F, sep="\t")
+         x <- read.table(fname, header=F, sep="\t")
```
[the remark this answers is not in the document]
Comment (pmpcosta) [reply tools/dada2/macros.xml:41]: ... return n

which tools do need version bump?

**Comment by pmpcosta** [comment]
dada2_uniques are generated by removeBimeraDenovo fed the output of dada, but give an error in seqCounts, because it is a vector and seqCounts is trying to do a rowSums with it... I would say it needs to be tested there, after it is fixed.

File: tools/dada2/macros.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (bernt-matthias) [reply tools/dada2/macros.xml:41]: > which tools do need version bump?
> we can do that here in the requirements section?

If a majority of the tools is affected I would just increase `WRAPPER_VERSION` in the macros file. Then all tools are updated.

**Comment by bernt-matthias** [comment]
> dada2_uniques are generated by removeBimeraDenovo fed the output of dada, but give an error in seqCounts, because it is a vector and seqCounts is trying to do a rowSums with it... I would say it needs to be tested there, after it is fixed.

Hmm, I think that we should include `dada2_uniques` here: 
https://github.com/galaxyproject/tools-iuc/blob/0382549abf85527cbdf01f36b3cab6a53648b76b/tools/dada2/dada2_seqCounts.xml#L24

You can use `removeBimeraDenovo_F3D0_dada_uniques.tab` as test input.

**Comment by bgruening** [comment]
@pmpcosta can you please look at @bernt-matthias last comment?

**Comment by pmpcosta** [comment]
Sorry, I have been busy lately, I will come back to it soon, I promise.

**Comment by bernt-matthias** [comment]
@pmpcosta I will include your commits over here https://github.com/galaxyproject/tools-iuc/pull/5430 

Thanks

**Pull request closed without merging**